Formula/f/fast-cli.rb

```diff
@@ -0,0 +1,35 @@
+class FastCli < Formula
+  desc "Command-line version of fast.com"
+  homepage "https://github.com/mikkelam/fast-cli"
+  url "https://github.com/mikkelam/fast-cli/archive/refs/tags/0.0.2.tar.gz"
+  sha256 "28b86b8c4cdefc86de4e0e124166cef010a7d1d00cdeee4d4beb17cab4717230"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GO111MODULE"] = "on"
+    dir = buildpath/"src/github.com/mikkelam/fast-cli"
+    dir.install buildpath.children
+
+    cd dir do
+      with_env(
+        "VERSION" => version.to_s,
+        "COMMIT"  => "homebrew",
+        "DATE"    => Time.now.utc.iso8601,
+      ) do
+        system "go", "mod", "tidy"
+        system "make", "build"
+      end
+      bin.install "fast-cli"
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/fast-cli --version")
+
+    output = shell_output("#{bin}/fast-cli --help")
+    assert_match "Estimate connection speed using fast.com", output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.